File: VB/CRUDBehavior/XPOServerModeCRUDBehavior.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Windows.Interactivity
Imports System.Windows
Imports DevExpress.Xpf.Grid
Imports System.Windows.Controls
Imports System.Windows.Input
Imports DevExpress.Xpf.Core
Imports System.ComponentModel
Imports DevExpress.Xpo
Imports CRUDBehaviorBase

Namespace XPOServer
    Public Class XPOServerModeCRUDBehavior
        Inherits CRUDBehaviorBase.CRUDBehaviorBase

        Protected Overrides Function CanExecuteRemoveRowCommand() As Boolean
            If CollectionSource Is Nothing OrElse Grid Is Nothing OrElse View Is Nothing OrElse View.FocusedRow Is Nothing Then
                Return False
            End If
            Return True
        End Function
        Protected Overrides Sub OnAttached()
            MyBase.OnAttached()
            If View IsNot Nothing AndAlso CollectionSource IsNot Nothing Then
                Initialize()
            Else
                AddHandler Grid.Loaded, AddressOf OnGridLoaded
            End If
        End Sub
        Protected Overrides Sub Initialize()
            Grid.ItemsSource = CollectionSource
            NewRowCommand.RaiseCanExecuteChangedEvent()
            RemoveRowCommand.RaiseCanExecuteChangedEvent()
            EditRowCommand.RaiseCanExecuteChangedEvent()
            MyBase.Initialize()
        End Sub
        Protected Overrides Sub UpdateDataSource()
            CollectionSource.Reload()
        End Sub
    End Class
End Namespace

File: CS/CRUDBehavior/XPOServerModeCRUDBehavior.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows.Interactivity;
using System.Windows;
using DevExpress.Xpf.Grid;
using System.Windows.Controls;
using System.Windows.Input;
using DevExpress.Xpf.Core;
using System.ComponentModel;
using DevExpress.Xpo;
using CRUDBehaviorBase;

namespace XPOServer {
    public class XPOServerModeCRUDBehavior: CRUDBehaviorBase.CRUDBehaviorBase {
        protected override bool CanExecuteRemoveRowCommand() {
            if(CollectionSource == null || Grid == null || View == null || View.FocusedRow == null) return false;
            return true;
        }
        protected override void OnAttached() {
            base.OnAttached();
            if(View != null && CollectionSource != null)
                Initialize();
            else
                Grid.Loaded += OnGridLoaded;
        }
        protected override void Initialize() {
            Grid.ItemsSource = CollectionSource;
            NewRowCommand.RaiseCanExecuteChangedEvent();
            RemoveRowCommand.RaiseCanExecuteChangedEvent();
            EditRowCommand.RaiseCanExecuteChangedEvent();
            base.Initialize();
        }
        protected override void UpdateDataSource() {
            CollectionSource.Reload();
        }
    }
}